src/utils/index.js

```diff
@@ -12,342 +12,342 @@ import nearestPointOnLine from "@turf/nearest-point-on-line";
 };
 
 export const addPointTovertices = (
-  map,
-  vertices,
-  coordinates,
-  forceInclusion
+    map,
+    vertices,
+    coordinates,
+    forceInclusion
 ) => {
-  const { width: w, height: h } = map.getCanvas();
-  // Just add verteices of features currently visible in viewport
-  const { x, y } = map.project(coordinates);
-  const pointIsOnTheScreen = x > 0 && x < w && y > 0 && y < h;
-
-  // But do add off-screen points if forced (e.g. for the current feature)
-  // So features will always snap to their own points
-  if (pointIsOnTheScreen || forceInclusion) {
-    vertices.push(coordinates);
-  }
+    const { width: w, height: h } = map.getCanvas();
+    // Just add verteices of features currently visible in viewport
+    const { x, y } = map.project(coordinates);
+    const pointIsOnTheScreen = x > 0 && x < w && y > 0 && y < h;
+
+    // But do add off-screen points if forced (e.g. for the current feature)
+    // So features will always snap to their own points
+    if (pointIsOnTheScreen || forceInclusion) {
+        vertices.push(coordinates);
+    }
 };
 
-export const createSnapList = (map, draw, currentFeature) => {
-  // Get all drawn features
-  const features = draw.getAll().features;
-  const snapList = [];
-
-  // Get current bbox as polygon
-  const bboxAsPolygon = (() => {
-    const canvas = map.getCanvas(),
-      w = canvas.width,
-      h = canvas.height,
-      cUL = map.unproject([0, 0]).toArray(),
-      cUR = map.unproject([w, 0]).toArray(),
-      cLR = map.unproject([w, h]).toArray(),
-      cLL = map.unproject([0, h]).toArray();
-
-    return bboxPolygon([cLL, cUR].flat());
-  })();
-
-  const vertices = [];
-
-  // Keeps vertices for drwing guides
-  const addVerticesTovertices = (coordinates, isCurrentFeature = false) => {
-    if (!Array.isArray(coordinates)) throw Error("Your array is not an array");
-
-    if (Array.isArray(coordinates[0])) {
-      // coordinates is an array of arrays, we must go deeper
-      coordinates.forEach((coord) => {
-        addVerticesTovertices(coord);
-      });
-    } else {
-      // If not an array of arrays, only consider arrays with two items
-      if (coordinates.length === 2) {
-        addPointTovertices(map, vertices, coordinates, isCurrentFeature);
-      }
-    }
-  };
-
-  features.forEach((feature) => {
-    // For currentfeature
-    if (feature.id === currentFeature.id) {
-      if (currentFeature.type === geojsonTypes.POLYGON) {
-        // For the current polygon, the last two points are the mouse position and back home
-        // so we chop those off (else we get vertices showing where the user clicked, even
-        // if they were just panning the map)
-        addVerticesTovertices(
-          feature.geometry.coordinates[0].slice(0, -2),
-          true
-        );
-      }
-      return;
-    }
+export const createSnapList = (map, draw, currentFeature, featuresList) => {
+    // Get all drawn features
+    const features = featuresList;
```

Comment:
Why this change is made when `featuresList` argument is never sent everywhere `createSnapList` is called?

Reply:
now the snapping doesn't even work!